fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexQueryPlan.java

```diff
@@ -161,28 +163,28 @@ public static LuceneIndexQueryPlan merge(LuceneIndexQueryPlan plan1, LuceneIndex
         final RecordMetaData metaData = store.getRecordMetaData();
         final Index index = metaData.getIndex(indexName);
         final Collection<RecordType> recordTypes = metaData.recordTypesForIndex(index);
-        if (recordTypes.size() != 1) {
-            throw new RecordCoreException("No lucene index should span multiple record types")
-                    .addLogInfo(LogMessageKeys.INDEX_NAME, indexName);
-        }
-
-        if (scanType == IndexScanType.BY_LUCENE_AUTO_COMPLETE) {
+        if (scanType == IndexScanType.BY_LUCENE_AUTO_COMPLETE || scanType == IndexScanType.BY_LUCENE_SPELLCHECK) {
+            if (recordTypes.size() != 1) {
+                throw new RecordCoreException("No lucene index should span multiple record types");
+            }
```

Comment:
any reason to move this check inside the if block for auto_complete and spell_check. I guess this check can be applied to ordinary lucene search as well?

Reply:
leftover from a commit merge.